### PR TITLE
Tick 351 simulate hawkes with non consta

### PR DESCRIPTION
**Pull request by Mbompr**

### CODE REVIEW

**Comment by Mbompr** [comment]
First commit is only code refactoring

File: tick/simulation/src/hawkes.h [hunk 1]
```diff
@@ -115,24 +75,41 @@ class Hawkes : public PP {
...
-  void set_mu(unsigned int i, const HawkesMuPtr &mu);
+  void set_baseline(unsigned int i, double baseline);
+
+  /**
+   * @brief Set baseline for a specific dimension
+   * \param i : the dimension
+   * \param time_function : a TimeFunction that will be used to construct a
+   * HawkesTimeFunctionBaseline
+   */
+  void set_baseline(unsigned int i, TimeFunction time_function);
```
Comment (vinther) [inline tick/simulation/src/hawkes.h]: Can be const-ref?

File: tick/simulation/src/hawkes.h [hunk 2]
```diff
@@ -115,24 +75,41 @@ class Hawkes : public PP {
...
+
+  /**
+  * @brief Set baseline for a specific dimension
+  * \param i : the dimension
+  * \param times : times that will be used to construct a HawkesTimeFunctionBaseline
+  * \param values : values that will be used to construct a HawkesTimeFunctionBaseline
+  */
+  void set_baseline(unsigned int i, ArrayDouble &times, ArrayDouble &values);
```
Comment (vinther) [inline tick/simulation/src/hawkes.h]: Can be const-ref?

Comment (Mbompr) [inline tick/simulation/src/hawkes.h]: Can't be as array[i] is not const

**Pull request merged**